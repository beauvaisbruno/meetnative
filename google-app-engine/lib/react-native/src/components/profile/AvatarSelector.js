"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const native_base_1 = require("native-base");
const React = __importStar(require("react"));
const react_native_1 = require("react-native");
const Avatars_1 = require("./img/Avatars");
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(
  require("react-native-auto-height-image")
);
const react_navigation_1 = require("react-navigation");
class AvatarSelector extends React.Component {
  componentWillUnmount() {
    this.navListener.remove();
  }
  componentDidMount() {
    this.navListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        if (
          this.props.navigation.state.params &&
          this.props.navigation.state.params.avatar
        ) {
          this.props.onChange(this.props.navigation.state.params.avatar);
          this.props.navigation.state.params.avatar = undefined;
        }
      }
    );
  }
  render() {
    return React.createElement(
      native_base_1.View,
      {
        style: {
          textAlignVertical: "top",
          justifyContent: "center",
          alignItems: "center"
        }
      },
      React.createElement(
        native_base_1.View,
        { style: { textAlignVertical: "top", flex: 1, flexDirection: "row" } },
        React.createElement(
          native_base_1.Text,
          { style: { textAlignVertical: "top", marginRight: 10 } },
          "Avatar"
        ),
        React.createElement(
          react_native_1.TouchableHighlight,
          {
            testID: "TouchableHighlight",
            onPress: () => {
              this.props.navigation.navigate("AvatarScreen");
            }
          },
          React.createElement(react_native_auto_height_image_1.default, {
            source: Avatars_1.avatars[this.props.value],
            width: react_native_responsive_screen_1.widthPercentageToDP("45%")
          })
        )
      )
    );
  }
}
exports.AvatarSelector = AvatarSelector;
exports.default = react_navigation_1.withNavigation(AvatarSelector);
//# sourceMappingURL=AvatarSelector.js.map
