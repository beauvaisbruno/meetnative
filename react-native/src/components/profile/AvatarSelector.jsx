"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var React = require("react");
var react_native_1 = require("react-native");
var Avatars_1 = require("./img/Avatars");
var react_native_responsive_screen_1 = require("react-native-responsive-screen");
var react_native_auto_height_image_1 = require("react-native-auto-height-image");
var react_navigation_1 = require("react-navigation");
var AvatarSelector = /** @class */ (function (_super) {
    __extends(AvatarSelector, _super);
    function AvatarSelector() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AvatarSelector.prototype.componentWillUnmount = function () {
        if (this.navListener)
            this.navListener.remove();
    };
    AvatarSelector.prototype.componentDidMount = function () {
        var _this = this;
        this.navListener = this.props.navigation.addListener("willFocus", function (payload) {
            if (_this.props.navigation.state.params &&
                _this.props.navigation.state.params.avatar) {
                _this.props.onChange(_this.props.navigation.state.params.avatar);
                _this.props.navigation.state.params.avatar = undefined;
            }
        });
    };
    AvatarSelector.prototype.render = function () {
        var _this = this;
        return (<native_base_1.View style={{
            textAlignVertical: "top",
            justifyContent: "center",
            alignItems: "center"
        }}>
        <native_base_1.View style={{ textAlignVertical: "top", flex: 1, flexDirection: "row" }}>
          <native_base_1.Text style={{ textAlignVertical: "top", marginRight: 10 }}>
            Avatar
          </native_base_1.Text>
          <react_native_1.TouchableHighlight testID={"TouchableHighlight"} onPress={function () {
            _this.props.navigation.navigate("AvatarScreen");
        }}>
            <react_native_auto_height_image_1.default source={Avatars_1.avatars[this.props.value]} width={react_native_responsive_screen_1.widthPercentageToDP("45%")}/>
          </react_native_1.TouchableHighlight>
        </native_base_1.View>
      </native_base_1.View>);
    };
    return AvatarSelector;
}(React.Component));
exports.AvatarSelector = AvatarSelector;
exports.default = react_navigation_1.withNavigation(AvatarSelector);
