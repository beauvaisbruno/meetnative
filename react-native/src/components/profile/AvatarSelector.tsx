import { View, Text } from "native-base";
import * as React from "react";
import { TouchableHighlight } from "react-native";
import { avatars } from "./img/Avatars";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";
import {
  NavigationEventSubscription,
  NavigationInjectedProps,
  withNavigation
} from "react-navigation";

export class AvatarSelector extends React.Component<
  {
    value: string;
    onChange: (avatar: string) => void;
  } & NavigationInjectedProps
> {
  navListener: NavigationEventSubscription | undefined;
  public componentWillUnmount() {
    if (this.navListener) this.navListener.remove();
  }
  public componentDidMount() {
    this.navListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        if (
          this.props.navigation.state.params &&
          this.props.navigation.state.params.avatar
        ) {
          this.props.onChange(this.props.navigation.state.params.avatar);
          this.props.navigation.state.params.avatar = undefined;
        }
      }
    );
  }

  public render() {
    return (
      <View
        style={{
          textAlignVertical: "top",
          justifyContent: "center",
          alignItems: "center"
        }}
      >
        <View
          style={{ textAlignVertical: "top", flex: 1, flexDirection: "row" }}
        >
          <Text style={{ textAlignVertical: "top", marginRight: 10 }}>
            Avatar
          </Text>
          <TouchableHighlight
            testID={"TouchableHighlight"}
            onPress={() => {
              this.props.navigation.navigate("AvatarScreen");
            }}
          >
            <AutoHeightImage
              source={avatars[this.props.value] as number}
              width={wp("45%")}
            />
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

export default withNavigation(AvatarSelector);
