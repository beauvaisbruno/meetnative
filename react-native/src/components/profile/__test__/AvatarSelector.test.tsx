jest.mock("react-navigation", () => {
  return {
    withNavigation: (param: any) => param
  };
});
import { shallow } from "enzyme";
import * as React from "react";
import toJson from "enzyme-to-json";
import { AvatarSelector } from "../AvatarSelector";

it("Avatar Selector", async () => {
  const navigate = jest.fn();
  const addListener = jest.fn((eventName, callback) => {
    callback("mockPayload");
  });
  const onChange = jest.fn();
  const tree = shallow(
    <AvatarSelector
      onChange={onChange}
      navigation={{
        navigate,
        // @ts-ignore
        addListener,
        // @ts-ignore
        state: { params: { avatar: "avatar" } }
      }}
    />
  );

  expect(toJson(tree)).toMatchSnapshot();
  expect(addListener.mock.calls[0][0]).toBe("willFocus");
  expect(onChange).toBeCalledWith("avatar");
  const TouchableHighlight = tree.findWhere(
    node => node.prop("testID") === "TouchableHighlight"
  );
  TouchableHighlight.props().onPress();

  expect(navigate).toBeCalledWith("AvatarScreen");
});
