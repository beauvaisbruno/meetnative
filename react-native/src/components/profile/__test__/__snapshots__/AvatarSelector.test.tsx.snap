// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Selector 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "textAlignVertical": "top",
    }
  }
>
  <Styled(ViewNB)
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "textAlignVertical": "top",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "marginRight": 10,
          "textAlignVertical": "top",
        }
      }
    >
      Avatar
    </Styled(Text)>
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      testID="TouchableHighlight"
      underlayColor="black"
    >
      <ErrorableImage
        width={337.5}
      />
    </TouchableHighlight>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
